pkg/service/updateBinary/updateBinary.go

```
@@ -0,0 +1,135 @@
+func (u *updater) UpdateBinary() {
+	currentVersion := utils.KeployVersion
+
+	// Fetch the latest version and release body from GitHub releases with a timeout
+	latestVersion, releaseBody, err := getLatestGitHubReleaseWithTimeout()
+
+	if err != nil {
+		if err == ErrGitHubAPIUnresponsive {
+			u.logger.Error("GitHub API is unresponsive. Update process cannot continue.")
+		} else {
+			u.logger.Error("Failed to fetch latest GitHub release version", zap.Error(err))
+		}
+		return
+	}
+
+	if currentVersion == latestVersion {
+		u.logger.Info("No updates available. Version " + latestVersion + " is the latest.")
+		return
+	}
+
+	u.logger.Info("Updating to Version: " + latestVersion)
+	// Execute the curl command to download keploy.sh and run it with bash
+	curlCommand := `curl -O https://raw.githubusercontent.com/keploy/keploy/main/keploy.sh && bash keploy.sh`
+
+	// Execute the combined curl command to download and execute keploy.sh with bash
+	cmd := exec.Command("sh", "-c", curlCommand)
+
+	// Set up input for the command
+	cmd.Stdin = os.Stdin
+
+	// Set output and error
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	// Start the command
+	if err := cmd.Start(); err != nil {
+		u.logger.Error("Failed to start command", zap.Error(err))
+		return
+	}
+
+	// Wait for command to finish
+	if err := cmd.Wait(); err != nil {
+		// Handle non-zero exit status here if required
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			u.logger.Error(fmt.Sprintf("Failed to update binary file: %v", exitErr))
+		} else {
+			u.logger.Error(fmt.Sprintf("Failed to wait for command: %v", err))
+		}
+		return
+	}
+
+	u.logger.Info("Updated Keploy binary to version " + latestVersion)
+	u.logger.Info("Release Notes:\n" + releaseBody)
+}
+
+// getLatestGitHubReleaseWithTimeout fetches the latest version and release body from GitHub releases with a timeout.
+func getLatestGitHubReleaseWithTimeout() (string, string, error) {
```

the timeout should be present in both the functions, Earlier I meant to say that the error returned in case of timeout will only be displayed in the update command, in other commands, we will simply ignore the error

---

also you can use githubRelease struct rather than having the two return values of type string

---

Understood, now the command will only throw the timeout error iin the update command